src/Shared/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyBannedPhraseChecksRule.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.BaseRules;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.Engine;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.Extensions;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.Rules.VacancyRules
+{
+    public sealed class VacancyBannedPhraseChecksRule : BaseBannedPhraseChecksRule, IRule<Vacancy>
+    {
+        private readonly IBannedPhrasesProvider _bannedPhrasesProvider;
+
+        public VacancyBannedPhraseChecksRule(
+            IBannedPhrasesProvider bannedPhrasesProvider, 
+            ConsolidationOption consolidationOption = ConsolidationOption.NoConsolidation, 
+            decimal weighting = 100.0m) 
+            : base(RuleId.BannedPhraseChecks, consolidationOption, weighting)
+        {
+            _bannedPhrasesProvider = bannedPhrasesProvider;
+        }
+
+        public async Task<RuleOutcome> EvaluateAsync(Vacancy subject)
```

Comment:
Should we get a test around this to ensure each property is being checked?

Reply:
I am going to merge as is for now as Sachin depends on this code to be checked in. I will add tests retrospectively.